### PR TITLE
Direct debit recaptcha validation, backend

### PR DESCRIPTION
**Pull request by tomrf1**
Adds a mandatory `recaptchaToken` field to `DirectDebitPaymentFields`.
The `CreateSubscriptionController` validates this token for all direct debit payment requests.
Reuses the `RecaptchaService`, currently used for validating creation of Stripe setup-intents.

### CODE REVIEW

File: support-frontend/app/controllers/CreateSubscriptionController.scala [hunk 1]
```diff
@@ -53,33 +58,77 @@ class CreateSubscriptionController(
...
-            } yield statusResponse
-
-            toHttpResponse(errorOrStatusResponse, request.body.product)
-        }
+        val maybeRecaptchaToken = getRecaptchaTokenFromRequest(request)
+
+        val errorOrStatusResponse = for {
+          _ <- maybeRecaptchaToken match {
+            case Some(token) => validateRecaptcha(token, testUsers.isTestUser(request))
+            case None => EitherT.rightT[Future, RequestValidationError](())
```
Comment (rupertbates) [inline support-frontend/app/controllers/CreateSubscriptionController.scala:66]: Won't this fail for all non-DD requests now?

**Review by rupertbates — APPROVED** [review]
Looks good to me 👍

File: support-frontend/app/controllers/CreateSubscriptionController.scala [hunk 2]
```diff
@@ -53,33 +58,77 @@ class CreateSubscriptionController(
...
-              _ <- validate(request)
-              supportWorkersUser = buildSupportWorkersUser(userAndEmail, request.body, testUsers.isTestUser(request))
-              statusResponse <- client
-                .createSubscription(request, supportWorkersUser, request.uuid)
-                .leftMap[CreateSubscriptionError](ServerError)
-            } yield statusResponse
-
-            toHttpResponse(errorOrStatusResponse, request.body.product)
-        }
+        val maybeRecaptchaToken = getRecaptchaTokenFromRequest(request)
```
Comment (rupertbates) [inline support-frontend/app/controllers/CreateSubscriptionController.scala:61]: Nitpick: Is this variable adding much vs inlining it?

File: support-frontend/app/controllers/CreateSubscriptionController.scala [hunk 3]
```diff
@@ -53,33 +58,77 @@ class CreateSubscriptionController(
...
+        toHttpResponse(errorOrStatusResponse, request.body.product)
       }
     }
 
+  private def getRecaptchaTokenFromRequest(
+      request: OptionalAuthRequest[CreateSupportWorkersRequest],
+  ): Option[String] = {
+    // Only Direct Debit requests include a recaptchaToken. For Stripe payments, recaptcha validation is done earlier in the flow
+    request.body.paymentFields match {
+      case Left(dd: DirectDebitPaymentFields) => Some(dd.recaptchaToken)
```
[the remark this answers is not in the document]
Comment (tomrf1) [reply support-frontend/app/controllers/CreateSubscriptionController.scala:79]: Correct, we're using the model + validation to enforce this.
Agree we need some thorough testing of this

File: support-frontend/app/controllers/CreateSubscriptionController.scala (hunk 2 above)
Comment (tomrf1) [reply support-frontend/app/controllers/CreateSubscriptionController.scala]: good point, and I always prefer to avoid `maybe` names!

**Pull request merged**